src/library/impl/meta_reader/database.h

```diff
@@ -57,19 +57,33 @@ namespace xlang::meta::reader
             {
                 return false;
             }
-
-            auto com = pe.OptionalHeader.DataDirectory[14]; // IMAGE_DIRECTORY_ENTRY_COM_DESCRIPTOR
-            auto sections = &file.as<impl::image_section_header>(dos.e_lfanew + sizeof(impl::image_nt_headers32));
+            
+            impl::image_section_header const* sections{};
+            uint32_t com_virtual_address{};
+            if (pe.OptionalHeader.Magic == 0x10B) // PE32
```

Comment:
Prefer to define these two values as constants. The metadata writer should share the value.

Reply:
Yes, should do that for the DOS magic number too